### DISCUSSION TITLE
Budibase Is there a way to add "None of the above" selection to Option Picker custom schema?

### DISCUSSION THREAD

**Discussion by mmays77** [Budibase/budibase#12492] [Help] [answered]
Budibase noob here, searching for some help that isn't explained in the docs. I am a self hosted community user running Budibase version 2.13.15 on Ubuntu 22.04 with Docker, recently installed Budibase.
I have an option picker that works fine. It is set up as a radio button. When my form/table loads as a fresh screen, I see all the rows in the table because the options picker has not been set, and all the rows with either "y" or "n" are displayed. My options picker is set to use a radio button for the user to choose between Business=y or Business=n and the rows are filtered on this character field having either a value of "y" or "n". When the user makes the selection the rows are filtered perfectly as expected, so the rows are selected and displayed to match the choice in the radio button. My question is this: how can I offer an option to remove the selection and go back to the "NULL" or "None of the above" or "BOTH" setting in the option picker and allow the user to see all the rows again without refreshing the page and starting over? Everything I have tried to set the "Both" value to results in no rows returned. 
 Thanks in advance.
![optionpicker01](https://github.com/Budibase/budibase/assets/152723073/f0edf4c1-e2e8-4a6a-9b94-f4ca8acaa69c)
![optionpicker02](https://github.com/Budibase/budibase/assets/152723073/4673f231-05cb-40e6-a1c1-697b369a5e54)
![optionpicker03](https://github.com/Budibase/budibase/assets/152723073/46009ade-6544-4218-8992-8cb2436a9cdc)
![optionpicker04](https://github.com/Budibase/budibase/assets/152723073/36d2b1b9-dd8f-4851-b960-c6178a31349d)

**Comment by ConorWebb96** [comment]
Hey @mmays77,

I think in this case all you need to do is add a value to both saying `both`. As for the `none of the above` option why not just include it with `N/A` or something along those lines after creating an additional select option?

That way whenever you go to filter the ones that are tagged with this in your database they won't appear in your results. I believe we have a few issues around [null](https://github.com/Budibase/budibase/issues/12210) elements at the moment.

I hope this makes sense.

**Reply by mmays77** [reply]
Thanks for your reply. I'm not sure I understand your suggestion. 

My intent is for the 3rd option to clear the selection, which would allow either of the other 2  options to be allowed. I could call it "Both" or "Clear" or "None of the above". I want it to have no selection and to display all the rows in the result set without any option set.

![image](https://github.com/Budibase/budibase/assets/152723073/f9b86bdd-0c7e-464b-83d4-f1b6f729941a)

For the 3rd value, I have tried emptying the field (which displays Value), "yn" "y|n" "None" "N/A" and as shown in the screenshot, {{ NULL }}. All of these return a set with no rows and the message "no rows found", which is the opposite of what I am hoping for, I want the 3rd selection to yield all rows, as happens when the screen is first displayed.

**Reply by mmays77** [reply]
I hoped that a dropdown options picker or a multi-select options picker would work. The dropdown options picker does not let you choose a value that is uninitialized as when the screen is refreshed. The multi-select filter cannot be used as a filter that allows multiple options to be selected. Several people have requested this as an enhancement but it keeps getting marked as stale.

**Comment by mmays77** [comment, marked as answer]
I found a workaround that resolves this problem. By changing the option type from "Radio Button" to "Select", and then keying in "Choose an option" in the placeholder field, the dropdown offers the custom options, such as "Y" or "N", and also has "Choose an option" in the list". This causes the selection to clear or reset to default, which in this case is the desired behavior.
